src/v/cluster/log_eviction_stm.cc

```diff
@@ -78,6 +78,10 @@ ss::future<> log_eviction_stm::write_raft_snapshots_in_background() {
               evict_until);
             try {
                 co_await do_write_raft_snapshot(*index_lb);
+            } catch (const ss::abort_requested_exception&) {
+                // ignore abort requested exception, shutting down
+            } catch (const ss::gate_closed_exception&) {
+                // ignore gate closed exception, shutting down
```

Comment:
Might be worth consider adding all from this list https://github.com/redpanda-data/redpanda/blob/dev/src/v/archival/ntp_archiver_service.cc#L227-L231, similar to ssx::spawn_with_gate.